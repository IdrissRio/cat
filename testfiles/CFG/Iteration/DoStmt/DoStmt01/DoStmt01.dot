digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[22]"; "Opt[25]"; "List[26]"; "List[27]"; "InstanceInitialization[6]"; "InstanceInitialization[14]"}
{ rank = same; "List[23]"; "MethodDecl[28]"; "Entry[7]"; "Exit[8]"; "Entry[15]"; "Exit[16]"}
{ rank = same; "Modifier[24]"; "Entry[29]"; "Modifiers[30]"; "PrimitiveTypeAccess[32]"; "List[33]"; "List[34]"; "Opt[35]"; "Exit[36]"}
{ rank = same; "List[31]"; "Block[36]"}
{ rank = same; "List[37]"}
{ rank = same; "VarDeclStmt[38]"; "DoStmt[47]"; "VarDeclStmt[59]"}
{ rank = same; "Modifiers[39]"; "PrimitiveTypeAccess[41]"; "List[42]"; "Block[48]"; "VarAccess[58]"; "Modifiers[60]"; "PrimitiveTypeAccess[62]"; "List[63]"}
{ rank = same; "List[40]"; "VariableDeclarator[43]"; "List[49]"; "List[61]"; "VariableDeclarator[64]"}
{ rank = same; "List[44]"; "Opt[45]"; "VarDeclStmt[50]"; "List[65]"; "Opt[66]"}
{ rank = same; "BooleanLiteral[46]"; "Modifiers[51]"; "PrimitiveTypeAccess[53]"; "List[54]"}
{ rank = same; "List[52]"; "VariableDeclarator[55]"}
{ rank = same; "List[56]"; "Opt[57]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < DoStmt01 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[22]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[24]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[25]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[27]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[28]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[29]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[30]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[31]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[32]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[33]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[34]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[35]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[36]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[37]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[38]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[39]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[40]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[41]" [label="PrimitiveTypeAccess\n < boolean >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[42]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[43]" [label="VariableDeclarator\n < a >", fillcolor=white   style=filled  ]
"List[44]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[45]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"BooleanLiteral[46]" [label="BooleanLiteral\n < false >", fillcolor=white   style=filled  ]
"DoStmt[47]" [label="DoStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[48]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[49]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[50]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[51]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[52]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[53]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[54]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[55]" [label="VariableDeclarator\n < b >", fillcolor=white   style=filled  ]
"List[56]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[57]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[58]" [label="VarAccess\n < a >", fillcolor=white   style=filled  ]
"VarDeclStmt[59]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[60]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[61]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[62]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[63]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[64]" [label="VariableDeclarator\n < c >", fillcolor=white   style=filled  ]
"List[65]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[66]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[36]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitialization[6]" [label="InstanceInitialization\n < Instance >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[7]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[8]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitialization[14]" [label="InstanceInitialization\n < Static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[15]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[16]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[23]" -> "Modifier[24]" [style=dashed, color=gray]
"Modifiers[22]" -> "List[23]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[22]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[25]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[26]" [style=dashed, color=gray]
"MethodDecl[28]" -> "Entry[29]" [style=dashed, color=gray]
"Modifiers[30]" -> "List[31]" [style=dashed, color=gray]
"MethodDecl[28]" -> "Modifiers[30]" [style=dashed, color=gray]
"MethodDecl[28]" -> "PrimitiveTypeAccess[32]" [style=dashed, color=gray]
"MethodDecl[28]" -> "List[33]" [style=dashed, color=gray]
"MethodDecl[28]" -> "List[34]" [style=dashed, color=gray]
"Modifiers[39]" -> "List[40]" [style=dashed, color=gray]
"VarDeclStmt[38]" -> "Modifiers[39]" [style=dashed, color=gray]
"VarDeclStmt[38]" -> "PrimitiveTypeAccess[41]" [style=dashed, color=gray]
"VariableDeclarator[43]" -> "List[44]" [style=dashed, color=gray]
"Opt[45]" -> "BooleanLiteral[46]" [style=dashed, color=gray]
"VariableDeclarator[43]" -> "Opt[45]" [style=dashed, color=gray]
"List[42]" -> "VariableDeclarator[43]" [style=dashed, color=gray]
"VarDeclStmt[38]" -> "List[42]" [style=dashed, color=gray]
"List[37]" -> "VarDeclStmt[38]" [style=dashed, color=gray]
"Modifiers[51]" -> "List[52]" [style=dashed, color=gray]
"VarDeclStmt[50]" -> "Modifiers[51]" [style=dashed, color=gray]
"VarDeclStmt[50]" -> "PrimitiveTypeAccess[53]" [style=dashed, color=gray]
"VariableDeclarator[55]" -> "List[56]" [style=dashed, color=gray]
"VariableDeclarator[55]" -> "Opt[57]" [style=dashed, color=gray]
"List[54]" -> "VariableDeclarator[55]" [style=dashed, color=gray]
"VarDeclStmt[50]" -> "List[54]" [style=dashed, color=gray]
"List[49]" -> "VarDeclStmt[50]" [style=dashed, color=gray]
"Block[48]" -> "List[49]" [style=dashed, color=gray]
"DoStmt[47]" -> "Block[48]" [style=dashed, color=gray]
"DoStmt[47]" -> "VarAccess[58]" [style=dashed, color=gray]
"List[37]" -> "DoStmt[47]" [style=dashed, color=gray]
"Modifiers[60]" -> "List[61]" [style=dashed, color=gray]
"VarDeclStmt[59]" -> "Modifiers[60]" [style=dashed, color=gray]
"VarDeclStmt[59]" -> "PrimitiveTypeAccess[62]" [style=dashed, color=gray]
"VariableDeclarator[64]" -> "List[65]" [style=dashed, color=gray]
"VariableDeclarator[64]" -> "Opt[66]" [style=dashed, color=gray]
"List[63]" -> "VariableDeclarator[64]" [style=dashed, color=gray]
"VarDeclStmt[59]" -> "List[63]" [style=dashed, color=gray]
"List[37]" -> "VarDeclStmt[59]" [style=dashed, color=gray]
"Block[36]" -> "List[37]" [style=dashed, color=gray]
"Opt[35]" -> "Block[36]" [style=dashed, color=gray]
"MethodDecl[28]" -> "Opt[35]" [style=dashed, color=gray]
"MethodDecl[28]" -> "Exit[36]" [style=dashed, color=gray]
"List[27]" -> "MethodDecl[28]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[27]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Entry[7]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Exit[8]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Entry[15]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Exit[16]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[14]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[16]" -> "Entry[15]" [color=red, constraint=false]
"Entry[15]" -> "Exit[16]" [color=blue, constraint=false, xlabel=" "] 
"Exit[8]" -> "Entry[7]" [color=red, constraint=false]
"Entry[7]" -> "Exit[8]" [color=blue, constraint=false, xlabel=" "] 
"Exit[36]" -> "VariableDeclarator[64]" [color=red, constraint=false]
"VariableDeclarator[64]" -> "Exit[36]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[64]" -> "VarAccess[58]" [color=red, constraint=false]
"VarAccess[58]" -> "VariableDeclarator[55]" [color=blue, constraint=false, xlabel="T "] 
"VarAccess[58]" -> "VariableDeclarator[64]" [color=blue, constraint=false, xlabel="F "] 
"VarAccess[58]" -> "VariableDeclarator[55]" [color=red, constraint=false]
"VariableDeclarator[55]" -> "VarAccess[58]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[55]" -> "VariableDeclarator[43]" [color=red, constraint=false]
"VariableDeclarator[55]" -> "VarAccess[58]" [color=red, constraint=false]
"VariableDeclarator[43]" -> "VariableDeclarator[55]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[43]" -> "BooleanLiteral[46]" [color=red, constraint=false]
"BooleanLiteral[46]" -> "VariableDeclarator[43]" [color=blue, constraint=false, xlabel=" "] 
"BooleanLiteral[46]" -> "Entry[29]" [color=red, constraint=false]
"Entry[29]" -> "BooleanLiteral[46]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"InstanceInitialization[14]" -> "InstanceInitialization[6]" -> "Modifiers[22]" -> "Opt[25]" -> "List[26]" -> "List[27]" [style=invis]
"Entry[29]" -> "Modifiers[30]" -> "PrimitiveTypeAccess[32]" -> "List[33]" -> "List[34]" -> "Opt[35]" -> "Exit[36]" [style=invis]
"VarDeclStmt[38]" -> "DoStmt[47]" -> "VarDeclStmt[59]" [style=invis]
"Modifiers[39]" -> "PrimitiveTypeAccess[41]" -> "List[42]" [style=invis]
"List[44]" -> "Opt[45]" [style=invis]
"Block[48]" -> "VarAccess[58]" [style=invis]
"Modifiers[51]" -> "PrimitiveTypeAccess[53]" -> "List[54]" [style=invis]
"List[56]" -> "Opt[57]" [style=invis]
"Modifiers[60]" -> "PrimitiveTypeAccess[62]" -> "List[63]" [style=invis]
"List[65]" -> "Opt[66]" [style=invis]
"Entry[7]" -> "Exit[8]" [style=invis]
"Entry[15]" -> "Exit[16]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
58", fillcolor=1]
Edges [label="Total Nr. Edges
75", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
9", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
9", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}