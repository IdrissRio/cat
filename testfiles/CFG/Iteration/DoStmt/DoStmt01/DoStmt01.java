public class DoStmt01 {
  void m() {
    boolean a = false;
    do {
      int b;
    } while (a);
    int c;
  }
}