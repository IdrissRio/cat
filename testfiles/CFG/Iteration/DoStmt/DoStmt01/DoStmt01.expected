NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[29] Successors: [BooleanLiteral[46]]
NodeName: BooleanLiteral[46] Successors: [VariableDeclarator[43]]
NodeName: VariableDeclarator[43] Successors: [VariableDeclarator[55]]
NodeName: VariableDeclarator[55] Successors: [VarAccess[58]]
NodeName: VarAccess[58] Successors: [VariableDeclarator[55], VariableDeclarator[64]]
NodeName: VariableDeclarator[64] Successors: [Exit[36]]
NodeName: Exit[36] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[36] Predecessor: [VariableDeclarator[64]]
NodeName: VariableDeclarator[64] Predecessor: [VarAccess[58]]
NodeName: VarAccess[58] Predecessor: [VariableDeclarator[55]]
NodeName: VariableDeclarator[55] Predecessor: [VariableDeclarator[43], VarAccess[58]]
NodeName: VariableDeclarator[43] Predecessor: [BooleanLiteral[46]]
NodeName: BooleanLiteral[46] Predecessor: [Entry[29]]
NodeName: Entry[29] Predecessor: []
